tuning/trainercontroller/operations/patience_hfcontrols.py

```diff
@@ -0,0 +1,56 @@
+# Standard
+from dataclasses import fields
+import inspect
+import re
+
+# Third Party
+from transformers import TrainerControl
+from transformers.utils import logging
+
+# Local
+from .operation import Operation
+
+logger = logging.get_logger(__name__)
+
+
+class HFControlsWithPatience(Operation):
```

Comment:
Why not enhance HFControls to support patience too, instead of adding a new class?

Reply:
Done.